core/src/macros.rs

```diff
@@ -260,4 +314,18 @@ mod tests {
         let hlist_pat![A, ...]    = hlist![A, B, C, D, E];
         let hlist_pat![A, B, ...] = hlist![A, B, C, D, E];
     }
+
+    #[test]
+    fn poly_fn_macro_test() {
+        use hlist::HMappable;
+        let h = hlist![9000, "joe", 41f32, "schmoe", 50];
+        let h2 = h.map(poly_fn!(
+            |x: i32| -> bool { x > 100 },
+            |x: f32| -> String { format!("{}", x) },
+            ['a] |x: &'a str| -> usize { x.len() }
```

Comment:
this probably needs more tests (e.g. at least a trailing comma after the last branch)

Reply:
Good point, done.